packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -539,6 +542,12 @@ class DeviceDomain extends Domain {
     if (!discoverer.supportsPlatform)
       return;
 
+    if (!discoverer.canListAnything)
+      sendEvent(
+          'daemon.showMessage',
+          'Unable to discover ${discoverer.name}. Please run "flutter doctor" '
```

Comment:
We may want to include a title and message field here. So, something like:

```
{
  'title': 'Unable to list devices',
  'message': 'Unable to list ${'discoverer.name}; please run "flutter doctor" to diagnose potential issues.'
}
```

or similar